### PR TITLE
Naughty Strings Test

### PR DESCRIPTION
**Pull request by justing-bq**
Implement Naughty Strings test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: naughty_strings_test.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/amzn/ion-go/ion"
+)
+
+func TestNaughtyStrings(t *testing.T) {
+	t.Skip() // Skipping test until ion text reader SymbolTable() is implemented
```
Comment (therapon) [inline naughty_strings_test.go:30]: Let's create an issue for each dependency we have on ion-hash that has made us skip test. I know this is at lest the second case, we had `IsStruct` and `SymbolTable`. Make an issue for each, please.

File: naughty_strings_test.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+	file, err := os.Open("ion-hash-test/big_list_of_naughty_strings.txt")
```
Comment (therapon) [inline naughty_strings_test.go:32]: Was the txt file also supposed to be in this PR? 
This txt file is a good candidate for a `test` or `_test_data`  folder that can hold data.

> Note that Go will also ignore directories or files that begin with "." or "_", so you have more flexibility in terms of how you name your test data directory.

Comment (justing-bq) [reply naughty_strings_test.go:32]: The txt file lives in the ion-hash-test folder which is a git submodule we added in https://github.com/amzn/ion-hash-go/pull/22/.

File: naughty_strings_test.go [hunk 3]
```diff
@@ -90,66 +92,43 @@ func TestNaughtyStrings(t *testing.T) {
...
-		if err != nil {
-			t.Fatalf("expected ionHashReader.Sum(nil) to execute without errors; %s", err.Error())
-		}
+		require.NoError(t, err, "Something went wrong executing ionHashReader.Sum(nil)")
 
-		if !reflect.DeepEqual(writerSum, readerSum) {
-			t.Errorf("expected reader/writer sums for \"%s\" to match;\nWriter sum: %v\nReader sum: %v",
-				tv.asIon(), writerSum, readerSum)
-		}
+		assert.Equalf(t, writerSum, readerSum, "Expected reader/writer sums for \"%s\" to match", s)
```
Comment (therapon) [inline naughty_strings_test.go:131]: What equality is the `assert` library using here? Is it the same as `reflect.DeepEqual`? 

I went to the repo for the `assert` lib but a quick look did not give me an answer.

**Pull request merged**